subsys/jwt/Kconfig

```diff
@@ -7,27 +8,78 @@ menuconfig JWT
 	help
 	  Enable creation of JWT tokens
 
+if JWT
+
 choice
 	prompt "JWT signature algorithm"
 	default JWT_SIGN_RSA
-	depends on JWT
 	help
 	  Select which algorithm to use for signing JWT tokens.
 
 config JWT_SIGN_RSA
 	bool "Use RSA signature (RS-256)"
-	depends on CSPRNG_ENABLED
-	select MBEDTLS
-	select MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
-	select PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY if PSA_CRYPTO_CLIENT
-	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT if PSA_CRYPTO_CLIENT
 
 config JWT_SIGN_ECDSA
 	bool "Use ECDSA signature (ES-256)"
+
+endchoice
+
+choice
+	default JWT_USE_PSA
+	prompt "Select crypto library to be used"
+
+config JWT_USE_PSA
+	bool "PSA crypto API library"
+	select MBEDTLS if !BUILD_WITH_TFM
+	select MBEDTLS_PSA_CRYPTO_C if !BUILD_WITH_TFM
+
+config JWT_USE_LEGACY
+	bool "Legacy library: TinyCrypt for ECDSA, Mbed TLS for RSA"
+
+endchoice
+
+# Hidden choice to select which algorithm and library will be used to sign
+# the JWT. User's selections on the above choices will determine which element
+# will be picked here.
+choice
+	prompt "Define which signature algorithm and crypto library is used to sign the JWT"
```

Comment:
I think that you want to just remove this `prompt` as well as those of the Kconfig options inside of this `choice`. They're not meant to have prompts, but to be automatically determined by the selections above, right?

Reply:
That's what I did initially, but then there are failures like:
`warning: the choice symbol JWT_SIGN_ECDSA_PSA (defined at subsys/jwt/Kconfig:47) has no prompt`

I agree that having a prompt makes little sense for these derived Kconfigs. I think it's better to go back to the previous implementation where these were not part of `choice`. Wdyt?

Reply:
Ah, damn. Well, whichever way then. In a sense you could even get completely rid of those extra Kconfig options and do all the logic in the PSA/LEGACY or RSA/ECDSA ones. But that would required having every `select` conditioned by an `if`, which is not great.